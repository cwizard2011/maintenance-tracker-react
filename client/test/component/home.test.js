import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import DefaultHome, { Home } from '../../src/components/Home';

const mockStore = configureMockStore([thunk]);

const props = {
  auth: false,
  user: {
    email: 'juliet',
    exp: 1540907850,
    iat: 1540886250,
    id: '48a698a0-1641-5aca-bc1b-de9b1a482ee1',
    user_role: 'user',
    username: 'juliet'
  }
};
const initialState = {
  authReducer: {
    user: {
      email: 'juliet',
      exp: 1540907850,
      iat: 1540886250,
      id: '48a698a0-1641-5aca-bc1b-de9b1a482ee1',
      user_role: 'user',
      username: 'juliet'
    },
    isAuthenticated: false
  }
};
const store = mockStore(initialState);

describe('Home component', () => {
  it('render home component correctly', () => {
    const wrapper = shallow(<Home {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render connected component', () => {
    const wrapper = shallow(<DefaultHome store={store} {...props} />);
    expect(wrapper.state()).toEqual({});
  });
  it('should Render user dashboard on login', () => {
    const newProps = {
      ...props,
      auth: true,
      history: {
        push: jest.fn()
      }
    };
    const wrapper = shallow(<Home {...newProps} />);
    expect(wrapper.state()).toEqual({});
  });
  it('should Render admin dashboard on login', () => {
    const newProps = {
      ...props,
      user: {
        user_role: 'admin'
      },
      auth: true,
      history: {
        push: jest.fn()
      }
    };
    const wrapper = shallow(<Home {...newProps} />);
    expect(wrapper.state()).toEqual({});
  });
});
