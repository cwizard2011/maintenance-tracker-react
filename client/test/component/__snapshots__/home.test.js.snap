// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component render home component correctly 1`] = `
<div>
  <HomeBody />
</div>
`;
